File: Byte&Buy.Backend/src/ByteBuy.Core/DTO/Offer/RentOffer/UserRentOfferResponse.cs
using ByteBuy.Core.DTO.Image;
using ByteBuy.Core.DTO.Money;

namespace ByteBuy.Core.DTO.Offer.RentOffer;

public record UserRentOfferResponse(
   Guid Id,
   Guid CategoryId,
   Guid ConditionId,
   string Name,
   string Description,
   int QuantityAvailable,
   MoneyDto PricePerDay,
   int MaxRentalDays,
   IReadOnlyCollection<ImageResponse> Images,
   IReadOnlyCollection<Guid>? ParcelLockerDeliveries,
   IReadOnlyCollection<Guid> OtherDeliveriesIds);

File: Byte&Buy.AngularSPA/src/app/features/offers/api-dto/user-rent-offer-response.ts
import { Guid } from "guid-typescript";
import { MoneyDto } from "../../../shared/api-dto/money-dto";
import { ImageResponse } from "../../../shared/api-dto/image-response";

export interface UserRentOfferResponse{
    id: Guid;
    categoryId: Guid;
    conditionId: Guid;
    name: string;
    description: string;
    quantityAvailable: number;
    pricePerDay: MoneyDto;
    maxRentalDays: number;
    images: ImageResponse[];
    parcelLockerDeliveries: Guid[] | null;
    otherDeliveriesIds: Guid[];
}